internal/edgeapi/backend/factory/factory.go

```diff
@@ -1,44 +1,88 @@
 package factory
 
 import (
+	"crypto/tls"
+	"github.com/project-flotta/flotta-operator/internal/edgeapi/backend"
+	"net/http"
+	"net/url"
+	"time"
+
 	"go.uber.org/zap"
 	"k8s.io/client-go/tools/record"
 	kubeclient "sigs.k8s.io/controller-runtime/pkg/client"
 
+	backendclient "github.com/project-flotta/flotta-operator/backend/client"
 	"github.com/project-flotta/flotta-operator/internal/common/repository/edgedevice"
 	"github.com/project-flotta/flotta-operator/internal/common/repository/edgedeviceset"
 	"github.com/project-flotta/flotta-operator/internal/common/repository/edgedevicesignedrequest"
 	"github.com/project-flotta/flotta-operator/internal/common/repository/edgeworkload"
 	"github.com/project-flotta/flotta-operator/internal/common/storage"
-	"github.com/project-flotta/flotta-operator/internal/edgeapi/backend"
 	"github.com/project-flotta/flotta-operator/internal/edgeapi/backend/k8s"
+	"github.com/project-flotta/flotta-operator/internal/edgeapi/backend/remote"
 	"github.com/project-flotta/flotta-operator/internal/edgeapi/configmaps"
 	"github.com/project-flotta/flotta-operator/internal/edgeapi/devicemetrics"
 	"github.com/project-flotta/flotta-operator/internal/edgeapi/images"
 	"github.com/project-flotta/flotta-operator/internal/edgeapi/k8sclient"
 )
 
-func Create(initialDeviceNamespace string, client kubeclient.Client, logger *zap.SugaredLogger, eventRecorder record.EventRecorder) backend.EdgeDeviceBackend {
+type Factory struct {
+	InitialDeviceNamespace string
+	Logger                 *zap.SugaredLogger
+	Client                 kubeclient.Client
+	EventRecorder          record.EventRecorder
+	TLSConfig              *tls.Config
+}
+
+func (f *Factory) Create(remoteBackendURL string, remoteBackendTimeout time.Duration) (backend.EdgeDeviceBackend, error) {
+	if remoteBackendURL == "" {
```

Comment:
This is too specific IMO, if we have to add new backend, we cannot only check some env; what will happen if some backend share the same parameters?
I would do something like that:
```

backendsMap := map[string]interface{}{
        "k8s":   f.createK8sBackend,
        "remote": f.createRemoteBackend,
    }
func ParseBackend(str string) (interface{}, bool) {
    c, ok := backendsMap[strings.ToLower(str)]
    return c, ok
}
```
(inspiration taken from [here](https://stackoverflow.com/questions/68543604/best-way-to-parse-a-string-to-an-enum))
And then you just call the `ParseBackend` with the (new) input argument explicitly selecting the backend.
I take as assumption that the method creating the backend will have a way of accessing the env var/arguments given to the program (so maybe we would need to add a config object that gather and give access to the env var needed to run)
WDYT?

Reply:
Hmm, being explicit about the backend choice has its appeal and I'll have a look how that would look like. I didn't feel strong about neither option.

Reply:
Please have a look now.